api/urls.py

```diff
@@ -27,6 +27,7 @@
 urlpatterns = [
     path('collections/', views.CollectionListView.as_view(), name='collections'),
     path('media/', views.MediaListView.as_view(), name='media'),
+    path('media/<media_key>', views.MediaView.as_view(), name='media'),
```

Comment:
This path has the same name as the one above it. (Goddamned words which don't change in the plural.) Perhaps rename the views to ``media_list`` and ``media_item``?